test/unit/memory/soa_vector.cpp

```diff
@@ -0,0 +1,249 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#include "test.hpp"
+#include <eve/memory/soa_vector.hpp>
+#include "unit/api/udt/udt.hpp"
+#include <memory>
+
+using t_t = kumi::tuple<int,float,char>;
+
+TTS_CASE("Check soa_vector default ctor")
+{
+  eve::soa_vector<t_t>          empty_tuple;
+  eve::soa_vector<udt::grid2d>  empty_udt;
+
+  TTS_EQUAL ( empty_tuple.size() , 0ULL );
+  TTS_EXPECT( empty_tuple.empty()       );
+
+  TTS_EQUAL ( empty_udt.size(), 0ULL  );
+  TTS_EXPECT( empty_udt.empty()       );
+}
+
+TTS_CASE("Check soa_vector sized ctor")
+{
+  eve::soa_vector<t_t>          tuple_vector( 37 );
+  eve::soa_vector<udt::grid2d>  udt_vector( 37 );
+
+  TTS_EQUAL     ( tuple_vector.size() , 37ULL );
+  TTS_EXPECT_NOT( tuple_vector.empty()        );
+
+  TTS_EQUAL ( udt_vector.size(), 37ULL);
+  TTS_EXPECT_NOT( udt_vector.empty()  );
+
+  // grid2d has some specific default
+  TTS_EQUAL( get<0>(udt_vector.get(0)), +1  );
+  TTS_EQUAL( get<1>(udt_vector.get(0)), -1  );
+}
+
+TTS_CASE("Check soa_vector sized ctor with default value")
+{
+  eve::soa_vector<t_t>          tuple_vector( 69, t_t{13,3.7f,'Z'} );
+  eve::soa_vector<udt::grid2d>  udt_vector  ( 69, udt::grid2d{4,20} );
```

Comment:
Nice :)

Not very professional though, I feel this is not a good look for a library to do this types of things.

Reply:
will change